### PR TITLE
github: public and legacy API changes require council review

### PR DESCRIPTION
**Pull request by DLehenbauer**
Configure CODEOWNERS so that deltas in the generated API reports automatically add 'fluid-cr-api' as a required reviewer.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -46,9 +46,13 @@
 /experimental/dds/tree/src                     @microsoft/fluid-cr-tree
 
 # Fluid Devtools source
-/packages/tools/devtools/**/src                   @microsoft/fluid-cr-devtools
+/packages/tools/devtools/**/src                @microsoft/fluid-cr-devtools
```
Comment (DLehenbauer) [inline .github/CODEOWNERS:49]: No real change here.  Just pedantically aligning the column with the other entries.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -46,9 +46,13 @@
...
+
+# Public and Legacy API changes
+/**/api-report/*.public.api.md                 @microsoft/fluid-cr-api
+/**/api-report/*.alpha.api.md                  @microsoft/fluid-cr-api
```
Comment (DLehenbauer) [inline .github/CODEOWNERS:53]: I think this is right.  Will test once merged.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -46,9 +46,13 @@
...
 
 # Package version files are not generally interesting to review by teams signing up for source tree ownership.
 # If a release team wanted to sign up to block on changes relevant for package release,
 # that would be the appropriate owner here. For now, explicitly give these files no owner.
-/**/packageVersion.ts
\ No newline at end of file
+/**/packageVersion.ts
```
Comment (DLehenbauer) [inline .github/CODEOWNERS:58]: Benign: vs code automatically added a trailing LF on save.  Not going to fight it.

File: .github/CODEOWNERS (hunk 2 above)
Comment (anthony-murphy) [reply .github/CODEOWNERS:53]: should we add beta as well? i don't think we want arbitrary things in beta.

**Review by anthony-murphy — APPROVED** [review]

**Review by Josmithr — APPROVED** [review]

**Comment by Abe27342** [comment]
FYI: Something I noticed in @jatgarg 's recent PR which changes alpha APIs is that api-extractor doesn't strip unused imports from  API reports. So changes to only the alpha API surface which introduce/remove certain imports can result in changes to public API report files. Hopefully this shouldn't happen too much so it's probably fine, but we should be aware this is what we're signing up for.

**Comment by Josmithr** [comment]
> FYI: Something I noticed in @jatgarg 's recent PR which changes alpha APIs is that api-extractor doesn't strip unused imports from API reports. So changes to only the alpha API surface which introduce/remove certain imports can result in changes to public API report files. Hopefully this shouldn't happen too much so it's probably fine, but we should be aware this is what we're signing up for.

I've been chatting with Pete (API-Extractor) about potential solutions for this (trimming the imports down to only the relevant ones for each report). We have an idea in mind that I'm hoping to get some time to prototype. I'll put something in the backlog to make sure it gets prioritized.

In the meantime, we _could_ patch our API-Extractor dep to completely omit import statements from reports, if we think that it is worth the loss of context the imports provide.

**Comment by anthony-murphy** [comment]
> In the meantime, we _could_ patch our API-Extractor dep to completely omit import statements from reports, if we think that it is worth the loss of context the imports provide.

@Josmithr  I like this plan, as the current imports are inaccurate, so i'm not sure what value they provide

**Comment by Josmithr** [comment]
> > In the meantime, we _could_ patch our API-Extractor dep to completely omit import statements from reports, if we think that it is worth the loss of context the imports provide.
> 
> @Josmithr I like this plan, as the current imports are inaccurate, so i'm not sure what value they provide

PR here: https://github.com/microsoft/FluidFramework/pull/21538

**Pull request merged**